sdk/src/main/java/com/mapbox/maps/MapView.kt

```diff
@@ -296,9 +300,45 @@ open class MapView : FrameLayout, MapPluginProviderDelegate, MapControllable {
    */
   @SuppressLint("ClickableViewAccessibility")
   override fun onTouchEvent(event: MotionEvent): Boolean {
+    interceptedActionList.firstOrNull()?.let {
+      try {
+        return mapController.onTouchEvent(it)
+      } finally {
+        it.recycle()
+        interceptedActionList.removeFirst()
+      }
+    }
     return mapController.onTouchEvent(event)
   }
 
+  override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
+    return when (event.actionMasked) {
+      MotionEvent.ACTION_DOWN -> {
+        interceptedActionList.add(MotionEvent.obtain(event))
+        false
+      }
+      MotionEvent.ACTION_POINTER_DOWN -> {
+        interceptedActionList.add(MotionEvent.obtain(event))
+        false
+      }
+      MotionEvent.ACTION_MOVE -> {
+        interceptedActionList.any { it.hypot(event) > touchSlop }
```

Comment:
I'm not sure this logic is correct. If I remember correctly we actually need to cache `ACTION_POINTER_DOWN` event just to explicitly pass it to gestures why threshold check here should check cached `ACTION_DOWN` with new event. Now it's quite hard for me to understand what happening - we're checking `any` element.

Could you please clarify what's the logic now? Using the list could actually accumulate events as we're polling only the first inside `onTouchEvent`.

Reply:
test with list size logging. this value does not exceed the number of fingers set on the screen when swiping.
suggest leave as is

Reply:
@alexander-kulikovskii to me it seems that for multi touch gestures we need to compare them by pointer id, since on multi touch gestures MotionEvent contain info about all the pointers. 
Moreover it seems we might not even need a list - just keep the latest event detected by ACTION_DOWN / ACTION_POINTER_DOWN, and then iterate through it's pointers, e.g. 
```
        for (i in 0 until event.pointerCount) {
// motion event contains info about all the pointers
          println(" id ${event.getPointerId(i)} / ${event.getX(i)} - ${event.getY(i)}")
        }

```

Reply:
good idea! 
refactor `hypot` function, but leave list of motion events. 
it's important to pass all events not only last one.